cheatsheets/Authentication_Cheat_Sheet.md

```diff
@@ -23,49 +23,24 @@ For information on validating email addresses, please visit the [input validatio
 
 A key concern when using passwords for authentication is password strength. A "strong" password policy makes it difficult or even improbable for one to guess the password through either manual or automated means. The following characteristics define a strong password:
 
-### Password Length
+- Password Length
+    - **Minimum** length of the passwords should be **enforced** by the application. Passwords **shorter than 8 characters** are considered to be weak ([NIST SP800-63B](https://pages.nist.gov/800-63-3/sp800-63b.html)). 
+    - **Maximum** password length should not be set **too low**, as it will prevent users from creating passphrases. Typical maximum length is 128 characters.
 
-Longer passwords provide a greater combination of characters and consequently make it more difficult for an attacker to guess.
+- Do do not truncate passwords. Make sure that every character the user types in is actually included in the password. 
 
-- **Minimum** length of the passwords should be **enforced** by the application.
-    - Passwords **shorter than 8 characters** are considered to be weak ([NIST SP800-63B](https://pages.nist.gov/800-63-3/sp800-63b.html)).
+- Allow usage of **all** characters including unicode and whitespaces. There should be no password composition rules limiting the type of characters permitted.
 
-While minimum length enforcement may cause problems with memorizing passwords among some users, applications should encourage them to set *passphrases* (sentences or combination of words) that can be much longer than typical passwords and yet much easier to remember.
+- Make sure that there are no periodic credential rotation or password history requirements.
 
-- **Maximum** password length should not be set **too low**, as it will prevent users from creating passphrases. Typical maximum length is 128 characters.
-    - Passphrases shorter than 20 characters are usually considered weak if they only consist of lower case Latin characters.
+- Include password strength meter to help users create a more complex password and block common and previously breached passwords
+    - [zxcvbn library](https://github.com/dropbox/zxcvbn) can be used for this purpose. (Note that this library is no longer maintained)
+    - [Pwned Passwords](https://haveibeenpwned.com/Passwords) is a service maintained by [Troy Hunt](https://www.troyhunt.com/) where passwords can be checked against previously breached passwords. You can host it yourself or use (API)(https://haveibeenpwned.com/API/v2#PwnedPasswords).
```

Comment:
I don't think that it is _still_ maintained by Troy? Recently I believe a company bought it from him. Just make sure about this information

Reply:
Indeed, it's seems the case, the service is searching for a new owner:
* https://threatpost.com/troy-hunt-sell-have-i-been-pwnd/145565/
* https://gizmodo.com/for-sale-have-i-been-pwned-1835413945
* https://techcrunch.com/2019/06/11/have-i-been-pwned-is-looking-for-a-new-owner

Reply:
Maybe simple remove that it is maintained by Troy?

Reply:
I cannot achieve to find if it was brought by someone but as it is in sell state I effectively think like you, that, we should remove it to prevent any "promotion" to a commercial company that we do not know...

Reply:
I thought only about removing "is a service maintained by [Troy Hunt](https://www.troyhunt.com/)" part

Reply:
Certainly. Doing that would save us the hassle of updating it. Just mentioned the service.